GVRf/Framework/framework/src/main/jni/objects/scene.cpp

```diff
@@ -23,9 +23,15 @@
 #include "objects/scene_object.h"
 
 namespace gvr {
+
+Scene* Scene::main_scene_ = NULL;
```

Comment:
One single main scene would be far better than the confusion we have now. I know there may be subtle errors which is why I programmed around them (I think). When the app launches main_scene will be the initial scene. When the app calls getNextMainScene the main_scene will be that one. When the app calls setMainScene the main_scene will be that one. I am trying to cover all the cases of when the app changes the scene.
GVRSceneObject.attachComponent will attach a collider component to the main scene. When the main scene changes, there is a one time overhead of gathering all the colliders from the scene graph and saving them in the "current" scene.
We need some way for the C++ layer to know the scene has changed so that it can do one-time initialization for the scene graph (like gathering the colliders).

I hate the whole getMainScene / getNextMainScene thing. This is silly. The app should make a GVRScene, attach their stuff and return that scene as the result of onInit. The view manager can then do what it needs to to make this the current scene.

The whole reason I need this collider list at all is to optimize the case where most of your objects are not pickable and to allow the possibility of picking stuff you can't see. This isn't important for the camera picking but when we start to use hand controllers it might be a case we encounter.